app/src/main/java/org/mozilla/fenix/settings/AccountSettingsFragment.kt

```diff
@@ -35,29 +34,24 @@ import org.mozilla.fenix.R
 import org.mozilla.fenix.components.metrics.Event
 import org.mozilla.fenix.ext.getPreferenceKey
 import org.mozilla.fenix.ext.requireComponents
-import kotlin.Exception
-import kotlin.coroutines.CoroutineContext
 
-class AccountSettingsFragment : PreferenceFragmentCompat(), CoroutineScope {
-    private lateinit var job: Job
-    override val coroutineContext: CoroutineContext
-        get() = Dispatchers.Main + job
+class AccountSettingsFragment : PreferenceFragmentCompat() {
     private lateinit var accountManager: FxaAccountManager
 
     // Navigate away from this fragment when we encounter auth problems or logout events.
     private val accountStateObserver = object : AccountObserver {
         override fun onAuthenticated(account: OAuthAccount) {}
 
         override fun onAuthenticationProblems() {
-            launch {
+            lifecycleScope.launch {
                 Navigation.findNavController(view!!).popBackStack()
```

Comment:
This is totally unrelated to your changes, but we should migrate to findNavController(fragment), which seems more resilient than hoping and praying the view is not null when we need it. https://developer.android.com/reference/androidx/navigation/fragment/NavHostFragment.html#findNavController(androidx.fragment.app.Fragment)